lang/java/reef-common/src/main/java/org/apache/reef/driver/catalog/ResourceCatalog.java

```diff
@@ -56,6 +56,13 @@
    */
   NodeDescriptor getNode(String nodeId);
 
+  /**
+   * We are not using this. It will be removed in the future. In order to do
+   * evaluator requests with specific rack names or node names, you should take
+   * a look at {@link EvaluatorRequest} new API.
+   *
+   */
+  @Deprecated
   public interface Descriptor {
 
     String getName();
```

Comment:
Is there a reason to deprecate this for the change at hand? If not, we should remove this change and file another JIRA for the implied API change.